datalad/cmdline/helpers.py

```diff
@@ -31,11 +32,25 @@ def __call__(self, parser, namespace, values, option_string=None):
             # lets use the manpage on mature systems ...
             try:
                 import subprocess
+                # extract version field from datalad manpage if present, swallow_error_logs
+                man_th = subprocess.check_output(
+                    ['zgrep', '\\.TH', '/usr/share/man/man1/datalad.1.gz'],
+                    stderr=subprocess.STDOUT)
+                man_version = man_th.split(' ')[5].strip(" '\"\t\n")
+
+                # extract current datalad version
+                datalad_version = '.'.join(
+                    __version__.split('.')[:3]
+                ).strip(" '\"\t\n")  # e.g 2.0.3.dev84 ~> 2.0.3
+
+                # don't show manpage if man_version not equal to current datalad_version
+                if datalad_version != man_version:
+                    raise ValueError
```

Comment:
and since this portion of the code is under `datalad/` it would actually be nice to have a unittest for it.
You could make use of that MANPATH env variable, to point to some custom path were you would generate a file with .TH for previous version and make sure that no subprocess.check_call  is called . and then with matching version and see that it is called with a correct `man` invocation